allennlp/modules/token_embedders/embedding.py

```diff
@@ -180,9 +188,33 @@ def extend_vocab(self,  # pylint: disable=arguments-differ
             vocab_namespace = "tokens"
             logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Defaulting to 'tokens'.")
 
+        extended_num_embeddings = extended_vocab.get_vocab_size(vocab_namespace)
+        if extended_num_embeddings <= self.num_embeddings:
+            # It's already been extended. No need to initialize / read pretrained file in first place (no-op)
+            return
+
+        if self._pretrained_file and os.path.exists(self._pretrained_file):
+            # pretrained_file attribute was saved during training and is available:
+            if pretrained_file:
+                # But user also passed pretrained_file
+                model_path_info = "at model_path, {model_path} " if model_path else ""
+                logging.warning(f"You passed pretrained_file, {pretrained_file} for "
+                                f"embedding {model_path_info}. But it's ignored because "
+                                f"original pretrained_file, {self._pretrained_file} is available.")
+            pretrained_file = self._pretrained_file
+
+        if not (pretrained_file and os.path.exists(pretrained_file)):
```

Comment:
I find checks written like this very hard to read / parse, I would prefer

```
if not pretrained_file or not os.path.exists(pretrained_file):
```

but as above I'm not sure the latter is the right check